index.js

```diff
@@ -107,55 +113,93 @@ exports = module.exports = function serveIndex(root, options){
     var dir = decodeURIComponent(url.pathname);
     var originalDir = decodeURIComponent(originalUrl.pathname);
 
-    // join / normalize from root dir
-    var path = normalize(join(root, dir));
+    // determine ".." display
+    var showUp = dir !== '/';
```

Comment:
I don't think so, as this is checking for `dir` not `originalDir`.
At least that was the result when I tested with connect, e.g. `app.use("/foo", serveIndex(dirs))`. You can not ".." to the root directory in this case.